File: java-backend/Processo.java
public class Processo implements Comparable<Processo>{
    
    private String pid;
    private int burstTime, arrivalTime;
    private String state;
    private int quantum;
    private int execTime;
    private int awaitTime;
    private int finishTime;

    public Processo(String pid, int burstTime, int arrivalTime, int quantum) {
        this.pid = pid;
        this.burstTime = burstTime;
        this.arrivalTime = arrivalTime;
        this.state = "Pronto";
        this.quantum = quantum;
        this.execTime = 0;
        this.awaitTime = 0;
        this.finishTime = 0;
    }

    public String toString(){
        String rep = "PID: " + this.pid +"\n";
        rep += "Tempo de chegada: " + this.arrivalTime + "\n";
        rep += "Burst Time: " + this.burstTime + "\n";

        if(this.state.equals("Finalizado")){
            rep += "Tempo de saida: " + this.finishTime + "\n";
            rep+= "Tempo de execucao: " + this.execTime + "\n";
            rep+= "Tempo de espera: " + this.awaitTime + "\n";
        }
        
        //rep += "Quantum: " + this.quantum + "\n"; 
        //rep += "State: " + this.state + "\n";
        return rep;
    }

    /* getters */
    public String getPid(){ return this.pid; } 
    public int getBurstTime(){ return this.burstTime; } 
    public int getArrivalTime(){ return this.arrivalTime; } 
    public String getState(){ return this.state; } 
    public int getQuantum(){ return this.quantum; } 
    public int getTempoExecucao(){ return this.execTime; }
    public int getAwaitTime(){ return this.awaitTime; }
    public int getFinishTime(){ return this.finishTime; }
    
    /* setters */
    public void setPid(String pid){
        this.pid = pid;
    }
    
    public void setBurstTime(int burstTime){
        this.burstTime = burstTime;
    }

    public void setArrivalTime(int arrivalTime){
        this.arrivalTime = arrivalTime;
    }

    private void setState(String state){
        this.state = state;
    }

    public void setQuantum(int quantum){
        this.quantum = quantum;
    }

    public void atualizaExecTime(){
        if(this.executando()){
            this.execTime = this.execTime + 1;
        }
    }

    public void resetaExecTime(){
        this.execTime = 0;
    }

    public void atualizaTempoEspera(){
        this.awaitTime = this.finishTime - this.arrivalTime - this.burstTime; 
    }

    private void setFinishTime(int finishTime){
        this.finishTime = finishTime;
    }

    // atualizadores do estado do processo
    public void finaliza(int tempoFinalizado){ 
        if(this.finalizado() == false){
            this.setState("Finalizado");
            this.setFinishTime(tempoFinalizado); 
            this.atualizaTempoEspera();
        }
    }
    
    public void bloqueia(){ this.setState("Bloqueado"); }
    public void prontifica(){ this.setState("Pronto"); }
    public void executa(){ this.setState("Executando"); }
    
    //checagem dos estados
    public boolean finalizado(){ return this.state.equals("Finalizado"); }
    public boolean bloqueado(){ return this.state.equals("Bloqueado"); }
    public boolean executando(){ return this.state.equals("Executando"); }
    public boolean pronto(){ return this.state.equals("Pronto"); }
    
    
    @Override
    public int compareTo(Processo p) {
        return (this.arrivalTime - p.arrivalTime);
    } 
}

File: ts-backend/src/Processo.ts
export enum processState{
    finalizado = 'Finalizado',
    pronto =  'Pronto',
    executando = 'Executando',
    bloqueado = 'Bloqueado',
    naoIniciado = 'NaoIniciado' 
}

export interface ProcessDto{
    pid: string,
    arrivalTime: number,
    burstTime: number,
    quantum: number
}

class Processo{
    private _pid: string;
    private _state: processState;
    private _burstTime: number;
    private _arrivalTime: number;
    private _quantum: number;
    private _execTime: number;
    private _finishTime: number;
    private _awaitTime: number;


    public constructor({pid, arrivalTime, burstTime, quantum}:ProcessDto){
        this._pid = pid;
        this._burstTime = burstTime;
        this._arrivalTime = arrivalTime;
        this._quantum = quantum;
        this._state = processState.naoIniciado;
        this._execTime = 0;
        this._finishTime = 0;
        this._awaitTime = 0;
    }
    //getters
    public get pid(){  return this._pid }
    public get state(){ return this._state }
    public get burstTime():number{ return this._burstTime }
    public get arrivalTime(){ return this._arrivalTime }
    public get quantum(){ return this._quantum }
    public get execTime(){ return this._execTime }
    public get finishTime(){ return this._finishTime }
    public get awaitTime(){ return this._awaitTime }
    
    //setters
    public set pid(pid: string){  this._pid  = pid }
    public set state(state: processState){ this._state = state }
    public set burstTime(burstTime: number){ this._burstTime = burstTime }
    public set arrivalTime(arrival_time: number){ this._arrivalTime = arrival_time }
    public set quantum(quantum: number){ this._quantum  = quantum}
    public set execTime(execTime: number){ this._execTime  = execTime}
    public set finishTime(finishTime: number){ this._finishTime = finishTime }
    public set awaitTime(awaitTime: number){ this._awaitTime = awaitTime }

    public bloqueia(){ this._state = processState.bloqueado; }
    public prontifica(){ this._state = processState.pronto; }
    public executa(){ this._state = processState.executando; }
    public removeDaCPU(){ this._state = processState.naoIniciado; }

    public finaliza(finishTime: number){ 
        if(this.finalizado() == false){
            this._state = processState.finalizado;
            this._finishTime = finishTime; 
            this.updateAwaitTime();
        }
    }

    public finalizado():boolean{ return this._state == processState.finalizado; }
    public bloqueado():boolean{ return this._state == processState.bloqueado; }
    public executando():boolean{ return this._state == processState.executando; }
    public pronto():boolean{ return this._state == processState.pronto; }
    public foraDaCPU():boolean{ return this._state == processState.naoIniciado; }

    public atualizaExecTime(){
        if(this.executando()){
            this.execTime = this.execTime + 1;
        }
    }

    public resetaExecTime(){
        this.execTime = 0;
    }

    public updateAwaitTime(){
        this._awaitTime = this.finishTime - this.arrivalTime - this.burstTime; 
    }
    
    public stringRepresentation() :string{
        let rep = "{ ";
        rep += "Pid: " + this._pid + ", ";
        rep += "Arrival Time: " + this._arrivalTime + ', ';
        rep += "Burst Time: " + this._burstTime + ', ';
        rep += "Total Execution Time: " + this._execTime + ', ' ;
        rep += "Quantum: " + this._quantum + ', ';
        rep+= "Process State: " + this._state;
        rep += " }"
        return rep;
    }
    
    public objStringRepresentation() :string{
        let rep = `{
            "pid": "${this.pid}",
            "arrivalTime": ${this._arrivalTime},
            "burstTime": ${this._burstTime},
            "quantum": ${this._quantum},
            "executionTime": ${this._execTime},
            "state": "${this._state}"
        }`;
        return rep;
    }

}

export default Processo